### PR TITLE
save metadata locally 

### PR DESCRIPTION
**Pull request by mroz22**
Changes here:
- renamed `UserDataProvider` to `FileSystemProvider`. It feels like more readable to me 
- finished implementation of read/write in electron layer. Note that I started to use `invoke` `handle` methods which (imho) should be used in this scenario instead off `on` `send` and `off` methods
- providers now have property `isCloud` to distinguish between cloud (dropbox, google) and local providers (local, maybe sd-card?). At the moment, this is only used to display different copy in the UI, but later there may be more differences between this types of providers such as one serving as general purpose primary provider and the other one (cloud-wise) as backup providers. this is to be discussed with product.
- providers `getCredentials` method was renamed to `getProviderDetails`. It does basically the same but also returns more information about the provider (`isCloud`) that we need to save into app storage. 
- FileSystem sync is enabled for desktop and is set behind feature flag. I believe that it is save to merge this and start testing it. The final decision whether this should be release lays with product.

### CODE REVIEW

File: packages/suite/src/actions/suite/__tests__/metadataActions.test.ts [hunk 1]
```diff
@@ -271,8 +273,9 @@ describe('Metadata Actions', () => {
         });
     });
 
-    fixtures.initMetadata.forEach(f => {
+    fixtures.init.forEach(f => {
         it(`initMetadata - ${f.description}`, async () => {
+            jest.mock('@suite/services/metadata/DropboxProvider');
```
Comment (szymonlesisz) [inline packages/suite/src/actions/suite/__tests__/metadataActions.test.ts]: `@suite-services`

File: packages/suite/src/actions/suite/metadataActions.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ import * as modalActions from '@suite-actions/modalActions';
 import * as notificationActions from '@suite-actions/notificationActions';
 import DropboxProvider from '@suite/services/metadata/DropboxProvider';
 import GoogleProvider from '@suite/services/metadata/GoogleProvider';
+import FileSystemProvider from '@suite/services/metadata/FileSystemProvider';
```
Comment (szymonlesisz) [inline packages/suite/src/actions/suite/metadataActions.ts]: `@suite-services`

File: packages/suite-desktop/src-electron/electron.ts [hunk 3]
```diff
@@ -12,6 +12,7 @@ import { buildMainMenu } from './menu';
 import { openBuyWindow } from './buy';
 // import * as metadata from './metadata';
```
Comment (mroz22) [inline packages/suite-desktop/src-electron/electron.ts]: useless import (comment)

**Comment by mroz22** [comment]
todo: 
- add changelog

**Review by szymonlesisz — APPROVED** [review]

**Pull request merged**